drivers/hmis_csv_importer/app/models/hmis_csv_importer/importer/importer.rb

```diff
@@ -80,8 +82,9 @@ def import!(import_log = nil)
       log_timing :cleanup_data_set!
       log_timing :analyze_tables
 
-      # determine what changes will be made and make note for alerting and monitoring
-      log_timing :precalculate_change_counts
+      # Determine what changes will be made and make note for alerting and monitoring. This is only needed if the data source is configured to pause on errors.
+      log_timing :precalculate_change_counts if @data_source.ever_pause_imports_with_errors?
```

Comment:
Thinking this through a bit more, I think this should be triggered if there are any import thresholds that would either send a notification OR pause the import.  
I'm good with this for now to prove it works, but before it gets merged we should update.  That will probably require a method or scope on `ImportThreshold` to indicate if it would ever pause or notify.

Reply:
good point, is this just an additional check for notification:
```suggestion
      log_timing :precalculate_change_counts if @data_source.ever_pause_imports_with_errors? || @data_source.ever_notify_for_imports?
```

Reply:
Yes!